libs/data/bbhnet/data/glitch_sampler.py

```diff
@@ -23,8 +23,17 @@ def to(self, device: str) -> None:
     def sample(
         self, N: int, size: int, trigger_distance_size: int = 0
     ) -> np.ndarray:
-        num_hanford = np.random.randint(N)
-        num_livingston = N - num_hanford
+        if N == 0:

```

Comment:
I _think_ this check is redundant, as it is checked [here](https://github.com/ML4GW/BBHNet/blob/main/libs/data/bbhnet/data/dataloader.py#L169). i.e., it's never even possible to call `glitch_sampler.sample` with `N = 0`.

Reply:
I don't think this function needed any changes for this PR. The only logic that needed adjusting is how the glitches are inserted into the background in the `__next__` function

Reply:
So I think you can remove a lot of those if else statements and have:

```
num_hanford = np.random.randint(N)
num_livingston = np.random.randint(N - num_hanford, N)
```